src/SprykerSdk/Zed/AopSdk/Business/OpenApi/Builder/OpenApiCodeBuilder.php

```diff
@@ -0,0 +1,902 @@
+            }
+            //Check for schema properties and recursively calling for instance of Reference
+            if ($schemaObject instanceof Reference && isset($schemaObject->properties) && !empty($schemaObject->properties)) {
+                $rootType[] = false;
+
+                return $this->getReponseParametersForReference($schemaObject, $rootType);
+            }
+
+            //Continue, if schemaObject does not have items
+            if (!isset($schemaObject->items)) {
+                continue;
+            }
+
+            //Check for schema's items properties and recursively calling for instance of Schema
+            if ($schemaObject->items instanceof Schema && isset($schemaObject->items->properties) && !empty($schemaObject->items->properties)) {
+                $rootType[] = true;
+
+                return $this->getReponseParametersForSchema($schemaObject->items, $rootType);
+            }
+            //Check for schema's items properties and recursively calling for instance of Reference
+            if ($schemaObject->items instanceof Reference && isset($schemaObject->items->properties) && !empty($schemaObject->items->properties)) {
+                $rootType[] = true;
+
+                return $this->getReponseParametersForReference($schemaObject->items, $rootType);
+            }
+        }
+        //Checks for, this method is calling first time or not
+        if (current($rootType) === true) {
+            //To set instances array to parsed data
+            //Example: [ .. "AppApiResponses" => "AppApiResponse[]:AppApiResponse" .. ]
+            $response = $this->setResponseParameterForReferenceClass($this->getClassNameFromReference($schema));
+        } else {
+            //To set array of key value pair to parsed data
+            //Example: [ .. "id" => "string" .. ]
+            $response = $this->setResponseParameterForKeyAndValue($this->getPropertiesFromReference($schema));
+        }
+
+        return $response;
+    }
+
+    /**
+     * To get array Reference classes
+     *
+     * @param string $className
+     *
+     * @return array
+     */
+    protected function setResponseParameterForReferenceClass(string $className): array
+    {
+        //Initialize array
+        $response = [];
+        //Replace 'Attributes' string with empty string
+        $refClass = str_replace('Attributes', '', $className);
+        //Pluralize and Assign reference class name to response
+        //Example: [ .. AppApiResponses => AppApiResponse[]:AppApiResponse ..]
+        $response[$this->inflector->pluralize($refClass)] = $refClass . '[]:' . $this->inflector->camelize($refClass);
+
+        return $response;
+    }
+
+    /**
+     * Set properties to parsed array of response
+     *
+     * @param iterable $properties
+     *
+     * @return array
+     */
+    protected function setResponseParameterForKeyAndValue(iterable $properties): array
```

Comment:
This method is not setting anything. `set` is mostly used to set a property of the class.

What is this for a response? The code looks like you build array specific transfer properties. Like:

```
$transferProperties = [
     'propertyName' => 'array[]:string',
];
``` 

TBH it's not really clear to me.

Your code is very hard to follow. You have a lot of functional methods that do "something" but the method names do not express what you are doing. Instead, you tried to explain with comments, which are 90% useless.

To get the class read and understandable organize it as follows:

```
public function build(OpenApiRequestTransfer $openApiRequestTransfer): OpenApiResponseTransfer
{
    $openApi = $this->load($openApiRequestTransfer->getTargetFileOrFail());
    $this->setSprykMode($openApiRequestTransfer);

    $openApiResponseTransfer = new OpenApiResponseTransfer();
    $openApiResponseTransfer = $this->generateTransfers($openApi, $openApiRequestTransfer, $openApiResponseTransfer);

    $openApiResponseTransfer = $this->generateXyz($openApi, $openApiRequestTransfer, $openApiResponseTransfer);

    return $openApiResponseTransfer;
}

protected function generateTransfers(OpenApi $openApi, OpenApiRequestTransfer $openApiRequestTransfer, OpenApiResponseTransfer $openApiResponseTransfer)
{
    $transferDefinitions = $this->getTransferDefinitions($openApi, $openApiRequestTransfer, $openApiResponseTransfer);
    $transferBuildSprykCommands = $this->getTransferDefinitionSprykCommands($transferDefinitions);

    $this->runCommands($transferBuildSprykCommands);

    return $openApiResponseTransfer;
}

protected function getTransferDefinitions(OpenApi $openApi, OpenApiRequestTransfer $openApiRequestTransfer, OpenApiResponseTransfer $openApiResponseTransfer)
{
    $transferDefinitions = [];

    foreach (...) {
        $moduleName = $this->getModuleName(...);
        $organizationName = $openApiRequestTransfer->getOrganizationNameOrFail();
        $transferName = $this->getTransferName(...);
        $transferProperties = $this->getTransferProperties(...);

        $transferDefinitions[$transferName] = [
            'moduleName' => $moduleName,
            'organizationName' => $organizationName,
            'transferName' => $transferName,
            'properties' => $transferProperties,
        ];
    }

    return $transferDefinitions;
}
```

This will make your code much more expressive. This is what the reader of your class is interested in. How you get the Transfer names and their properties is below of these methods but should also as clear as possible.